basesandbox/src/ipc/domain_socket.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+
+use super::Ipc;
+use std::os::unix::net::UnixDatagram;
+use std::path::Path;
+
+pub struct DomainSocket {
+    address_src: String,
+    address_dst: String,
+    socket: UnixDatagram,
+    buffer: Vec<u8>,
+}
+
+fn create(address_src: String, address_dst: String) -> DomainSocket {
+    DomainSocket {
```

Comment:
The current code unnecessarily clones the `address_src`.
Of course, the address_src would not be a big object and cloning might not harm the performance, but it still uses the global heap resource because of the String implementation.
You can rewrite the code if you create a socket first because [`UnixDatagram::bind()`](https://doc.rust-lang.org/std/os/unix/net/struct.UnixDatagram.html#method.bind) takes `AsRef<Path>`.

```rust
    let socket = UnixDatagram::bind(&address_src).unwrap();
    DomainSocket {
        address_src,
        address_dst,
        socket,
        buffer: vec![0; 1024],
    }
```